regipy/plugins/system/network_data.py

```diff
@@ -30,49 +28,52 @@ def get_network_info(self, subkey, interfaces=None):
-                        "ip_address": interface.get_value("IPAddress"),
-                        "subnet_mask": interface.get_value("SubnetMask"),
-                        "default_gateway": interface.get_value("DefaultGateway"),
-                        "name_server": interface.get_value("NameServer"),
-                        "domain": interface.get_value("Domain"),
-                    }
-                )
-
-            if interface.subkey_count:
-                sub_interfaces = []
-                self.get_network_info(self, interface, sub_interfaces)
-                entries["sub_interface"] = sub_interfaces
+            try:
+                if entries["dhcp_enabled"]:
+                    entries.update(
+                        {
+                            "dhcp_server": interface.get_value("DhcpServer"),
+                            "dhcp_ip_address": interface.get_value("DhcpIPAddress"),
+                            "dhcp_subnet_mask": interface.get_value("DhcpSubnetMask"),
+                            "dhcp_default_gateway": interface.get_value("DhcpDefaultGateway"),
+                            "dhcp_name_server": interface.get_value("DhcpNameServer"),
+                            "dhcp_domain": interface.get_value("DhcpDomain"),
+                        }
+                    )
+
+                    lease_obtained_time = interface.get_value("LeaseObtainedTime")
+                    if lease_obtained_time is not None:
+                        lease_obtained_time_str = datetime.fromtimestamp(
+                            lease_obtained_time, timezone.utc
+                        ).strftime("%Y-%m-%d %H:%M:%S")
+                        entries["dhcp_lease_obtained_time"] = lease_obtained_time_str
+
+                    lease_terminates_time = interface.get_value("LeaseTerminatesTime")
+                    if lease_terminates_time is not None:
+                        lease_terminates_time_str = datetime.fromtimestamp(
+                            lease_terminates_time, timezone.utc
+                        ).strftime("%Y-%m-%d %H:%M:%S")
+                        entries["dhcp_lease_terminates_time"] = lease_terminates_time_str
+
+                else:
+                    entries.update(
+                        {
+                            "ip_address": interface.get_value("IPAddress"),
+                            "subnet_mask": interface.get_value("SubnetMask"),
+                            "default_gateway": interface.get_value("DefaultGateway"),
+                            "name_server": interface.get_value("NameServer"),
+                            "domain": interface.get_value("Domain"),
+                        }
+                    )
+
+                if interface.subkey_count:
+                    sub_interfaces = []
+                    sub_interfaces = self.get_network_info(interface, sub_interfaces)
+                    entries["sub_interface"] = sub_interfaces
+
+            except Exception as e:
```

Comment:
Is there a specific issue that is being handled here? It's better to return from the plugin as much data as possible, possibly with an additional key indicating that it is incomplete. 
In the close future when I refactor plugin output we'll be able to handle that there.